### Understand the data properly

We are not plotting today.
We are thinking.



In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

# Before you start make sure you have installed matplotlib and seaborn

In [3]:
data = pd.read_csv("vgsales.csv")

### Step 1️ — Basic Structure

Run:

```python
df.shape
df.info()
df.head()
df.tail()
```

You need to answer:

* How many rows?
* How many columns?
* Any missing values?
* Which columns are float?
* Which are object (categorical)?

For video game sales typically:

* Categorical → Name, Platform, Genre, Publisher
* Numerical → Year, NA_Sales, EU_Sales, Global_Sales etc.

If you don’t know column types, your plots will be wrong.

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16598 entries, 0 to 16597
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Rank          16598 non-null  int64  
 1   Name          16598 non-null  object 
 2   Platform      16598 non-null  object 
 3   Year          16327 non-null  float64
 4   Genre         16598 non-null  object 
 5   Publisher     16540 non-null  object 
 6   NA_Sales      16598 non-null  float64
 7   EU_Sales      16598 non-null  float64
 8   JP_Sales      16598 non-null  float64
 9   Other_Sales   16598 non-null  float64
 10  Global_Sales  16598 non-null  float64
dtypes: float64(6), int64(1), object(4)
memory usage: 1.4+ MB


### Step 2️ — Identify Variable Types Clearly

###  Numerical Columns

These usually include:

* Year
* Sales columns
* Critic score (if exists)
* User score

Now ask:

* Are they integers or floats?
* Are there weird values like 0 year or 999?

---

### Categorical Columns

Examples:

* Genre
* Platform
* Publisher

Ask:

* How many unique categories?

```python
df['Genre'].nunique()
df['Genre'].value_counts()
```

If a column has 500 unique values (like Name), it's not useful for bar charts directly.

---

In [5]:
data["Genre"].nunique()
data["Genre"].unique()

array(['Sports', 'Platform', 'Racing', 'Role-Playing', 'Puzzle', 'Misc',
       'Shooter', 'Simulation', 'Action', 'Fighting', 'Adventure',
       'Strategy'], dtype=object)

In [6]:
data["Genre"].value_counts()

Genre
Action          3316
Sports          2346
Misc            1739
Role-Playing    1488
Shooter         1310
Adventure       1286
Racing          1249
Platform         886
Simulation       867
Fighting         848
Strategy         681
Puzzle           582
Name: count, dtype: int64

### Step 3 — Descriptive Statistics (Critical Step)

Run:

```python
df.describe()
```

This gives you:

* Count
* Mean
* Standard deviation
* Minimum
* 25th percentile (Q1)
* 50th percentile (Median)
* 75th percentile (Q3)
* Maximum

Now don’t just stare at numbers. Interpret them.

---

### 1️ Mean vs Median — Detect Skewness

Compare:

* **Mean**
* **50% (Median)**

### If:

* Mean > Median → Right skewed
* Mean < Median → Left skewed

In video game sales data, most sales columns are **right skewed**.

Why?

Because:

* Most games sell small numbers
* A few blockbuster games sell extremely high

That creates a long right tail.

Example:

* Mean = 0.5 million
* Median = 0.15 million
* Max = 82 million

That’s heavy right skew.

This tells you:
Most games sell very little. A few dominate the market.

That’s real insight.

---

### Minimum and Maximum — Reality Check

Look at:

* Minimum
* Maximum

Ask yourself:

* Is the minimum reasonable?
* Is the maximum extremely large compared to mean?

If:

* Max is 100x bigger than median
* Or standard deviation is very high

Then the distribution is spread out heavily.

That signals outliers or extreme variance.

---

### 3️ Standard Deviation — Spread of Data

Standard deviation tells you how much values deviate from the mean.

If standard deviation is large compared to the mean:

That means data is highly dispersed.

In sales datasets, this is common.

Why?

Because:

* Most games cluster near small values
* A few explode in sales

High spread = high variability in performance.

---

### 4️ Are Most Values Near Zero?

Even before plotting, think:

* Is Q1 very close to 0?
* Is median small?
* Is max extremely high?

If yes:

Most data is clustered near lower values, and only a few are very high.

That’s classic long-tail distribution.

---

### Outliers Are Not Always Errors

In video game sales:

Blockbusters are expected outliers.

Examples of legitimate extreme sellers:

* Grand Theft Auto V
* Super Mario Bros.
* Wii Sports

These are not data mistakes.
They are real-world dominance.

Later, when you plot a boxplot, these will show up as extreme points.

Don’t panic. That’s expected.

---

### What You Should Conclude After `describe()`

You should be able to say something like:

* “Global sales are highly right skewed.”
* “Most games sell under 1 million.”
* “A few games dominate total sales.”
* “There is high variability in performance.”

If you can say that confidently, you are thinking like a data analyst.

If you’re just reading numbers without conclusions, you’re wasting time.

---


In [8]:
data.describe()

,Rank,Year,NA_Sales,EU_Sales,JP_Sales,Other_Sales,Global_Sales
count,16598.000000,16327.000000,16598.000000,16598.000000,16598.000000,16598.000000,16598.000000
mean,8300.605254,2006.406443,0.264667,0.146652,0.077782,0.048063,0.537441
std,4791.853933,5.828981,0.816683,0.505351,0.309291,0.188588,1.555028
min,1.000000,1980.000000,0.000000,0.000000,0.000000,0.000000,0.010000
25%,4151.250000,2003.000000,0.000000,0.000000,0.000000,0.000000,0.060000
50%,8300.500000,2007.000000,0.080000,0.020000,0.000000,0.010000,0.170000
75%,12449.750000,2010.000000,0.240000,0.110000,0.040000,0.040000,0.470000
max,16600.000000,2020.000000,41.490000,29.020000,10.220000,10.570000,82.740000


### Interpretations

#### NA_Sales

* Mean = 0.264
* Median = 0.08
* Std = 0.816
* Max = 41.49

#### Interpretation:

Mean is much higher than median → clear right skew.

Most games sell under 0.1 million in North America.

The 75th percentile is 0.24, meaning even the top 25% of games don’t cross 0.25 million by much.

But the maximum is 41.49 million — extreme blockbuster effect.

Standard deviation (0.816) is 3x the mean. That’s massive spread.

Conclusion:
North America is dominated by a few massive hits while the majority of games perform poorly.

This is strongly right skewed.

---

#### EU_Sales

* Mean = 0.146
* Median = 0.02
* Std = 0.505
* Max = 29.02

Median is extremely small (0.02).
Mean is much larger.

That gap screams right skew.

75% of games sell under 0.11 million.

Yet the maximum hits 29 million.

Standard deviation is 3.4x the mean.

Conclusion:
Europe also follows blockbuster dominance, but overall average sales are lower than NA.

Very heavy right skew.

---

#### JP_Sales

* Mean = 0.077
* Median = 0
* Std = 0.309
* Max = 10.22

Median is literally 0.

That means more than half the games sold almost nothing in Japan.

Yet some games go up to 10 million.

Standard deviation is 4x the mean.

Conclusion:
Japan market is even more concentrated. Very few games perform strongly. Majority have negligible sales.

Extreme right skew.

---

#### Other_Sales

* Mean = 0.048
* Median = 0.01
* Std = 0.188
* Max = 10.57

Same pattern again.

Median very small.
Max very high relative to mean.

Conclusion:
Rest of the world market behaves like Japan — most games sell little, few explode.

Right skewed.

---

#### Global_Sales

* Mean = 0.537
* Median = 0.17
* Std = 1.55
* Max = 82.74

Mean is 3x the median.

Standard deviation is almost 3x the mean.

Maximum is enormous compared to central tendency.

Conclusion:
The global market is highly unequal.
Few games dominate global revenue.
Most games are low-performing.

Very strong right skew.

---


### Step 4 — Missing Values

Check:

```python
df.isnull().sum()
```

Ask:

* Is missing data random?
* Or concentrated in specific columns?

For example:
If Year has missing → You can’t plot time trends properly.

---

In [9]:
data.isna().sum()

Rank              0
Name              0
Platform          0
Year            271
Genre             0
Publisher        58
NA_Sales          0
EU_Sales          0
JP_Sales          0
Other_Sales       0
Global_Sales      0
dtype: int64

In [10]:
data.isnull().sum()

Rank              0
Name              0
Platform          0
Year            271
Genre             0
Publisher        58
NA_Sales          0
EU_Sales          0
JP_Sales          0
Other_Sales       0
Global_Sales      0
dtype: int64